### DISCUSSION TITLE
rAudio 20221208: Existing bookmarks unusable

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1128] [Fixed Issues]
This is my (minimal) library configuration:
<img src="https://user-images.githubusercontent.com/25383513/206684628-1d19a019-1e12-417e-9ab1-adab9f1f1e37.jpg" width=300>
In particular: "Network" is unchecked.

After the update to 20221208 the library view looks like this:
<img src="https://user-images.githubusercontent.com/25383513/206686262-78275386-c566-4a35-bc37-607261c7175c.jpg" width=600>
You see, the coverarts of the bookmarks are gone.

If I press one of the bookmarks I get:
<img src="https://user-images.githubusercontent.com/25383513/206686953-fb389445-952a-4e7f-8f80-c60e212a6138.jpg" width=600>
You see: nothing!

No chance to traverse the library starting with bookmarks. As a workaround I have to check the "Network" in the library configuration to traverse the library via "NAS", ignoring the bookmarks.

**Comment by rern** [comment]
@klassikmann,
The update removes thumbnail paths from each bookmark files which yours still exist.
```sh
readarray -t bookmarks <<< $( ls -1 /srv/http/data/bookmarks/* )
if [[ $bookmarks ]]; then
	for file in "${bookmarks[@]}"; do
		(( $( wc -l < "$file" ) > 1 )) && sed -i '2,$ d' "$file"
 	done
fi
```
Try the script. If there're no errors, each bookmark file should contain only a single.

**Reply by klassikmann** [reply]
@rern 
All of my existing bookmark files contain two lines but the second line does not end with a '\n'. Consequence: "wc -l" delivers 1 not 2. So the script is not working. But after appending a '\n' to all bookmark files, then applying your script solves the problem.
